external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -129,11 +134,15 @@ impl PartialEq for Package {
         // comparing packages during insertion of externally resolved ones - an internally resolved
         // existing package in the graph would not be recognized as a potential different version of
         // the externally resolved one)
-        self.kind == other.kind
+        match (&self.version, &other.version) {
```

Comment:
This use of `version` made me think that this name is not doing the field justice -- it's being used for a lot more than just a version: It overrides the identity of the package as a whole. Is it worth giving it a more generic name accordingly?

Maybe I'm missing something, but it also makes the test runner setup a little strange, because currently it equates all packages that have the same `version` field. I think that's fine today because all packages were given different versions, but in practice, maybe the test-runner should implement `resolve_version` by creating some kind of combined string that represents both the package name and its version field?  Giving the resolver and field a more generic name would make that kind of thing easier to spot as well.